src/app/qgisapp.cpp

```diff
@@ -646,6 +651,13 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, bool skipVersionCh
   sInstance = this;
   QgsRuntimeProfiler *profiler = QgsApplication::profiler();
 
+  startProfile( QStringLiteral( "User profile manager" ) );
+  mUserProfileManager = new QgsUserProfileManager();
```

Comment:
mUserProfileManager is never deleted. Can you make it a unique_ptr or parent it to the app instance?